torch/csrc/Exceptions.cpp

```diff
@@ -53,87 +55,82 @@ could not be completed because the input matrix is singular.",
+          {"Variable[SparseCPUShortType]", "torch.sparse.ShortTensor"},
+          {"Variable[SparseCPUHalfType]", "torch.sparse.HalfTensor"},
+          {"Variable[CUDAByteType]", "torch.cuda.ByteTensor"},
+          {"Variable[CUDACharType]", "torch.cuda.CharTensor"},
+          {"Variable[CUDADoubleType]", "torch.cuda.DoubleTensor"},
+          {"Variable[CUDAFloatType]", "torch.cuda.FloatTensor"},
+          {"Variable[CUDAIntType]", "torch.cuda.IntTensor"},
+          {"Variable[CUDALongType]", "torch.cuda.LongTensor"},
+          {"Variable[CUDAShortType]", "torch.cuda.ShortTensor"},
+          {"Variable[CUDAHalfType]", "torch.cuda.HalfTensor"},
+          {"Variable[CPUByteType]", "torch.ByteTensor"},
+          {"Variable[CPUCharType]", "torch.CharTensor"},
+          {"Variable[CPUDoubleType]", "torch.DoubleTensor"},
+          {"Variable[CPUFloatType]", "torch.FloatTensor"},
+          {"Variable[CPUIntType]", "torch.IntTensor"},
+          {"Variable[CPULongType]", "torch.LongTensor"},
+          {"Variable[CPUShortType]", "torch.ShortTensor"},
+          {"Variable[CPUHalfType]", "torch.HalfTensor"},
+          {"SparseCUDAByteType", "torch.cuda.sparse.ByteTensor"},
+          {"SparseCUDACharType", "torch.cuda.sparse.CharTensor"},
+          {"SparseCUDADoubleType", "torch.cuda.sparse.DoubleTensor"},
+          {"SparseCUDAFloatType", "torch.cuda.sparse.FloatTensor"},
+          {"SparseCUDAIntType", "torch.cuda.sparse.IntTensor"},
+          {"SparseCUDALongType", "torch.cuda.sparse.LongTensor"},
+          {"SparseCUDAShortType", "torch.cuda.sparse.ShortTensor"},
+          {"SparseCUDAHalfType", "torch.cuda.sparse.HalfTensor"},
+          {"SparseCPUByteType", "torch.sparse.ByteTensor"},
+          {"SparseCPUCharType", "torch.sparse.CharTensor"},
+          {"SparseCPUDoubleType", "torch.sparse.DoubleTensor"},
+          {"SparseCPUFloatType", "torch.sparse.FloatTensor"},
+          {"SparseCPUIntType", "torch.sparse.IntTensor"},
+          {"SparseCPULongType", "torch.sparse.LongTensor"},
+          {"SparseCPUShortType", "torch.sparse.ShortTensor"},
+          {"SparseCPUHalfType", "torch.sparse.HalfTensor"},
+          {"CUDAByteType", "torch.cuda.ByteTensor"},
+          {"CUDACharType", "torch.cuda.CharTensor"},
+          {"CUDADoubleType", "torch.cuda.DoubleTensor"},
+          {"CUDAFloatType", "torch.cuda.FloatTensor"},
+          {"CUDAIntType", "torch.cuda.IntTensor"},
+          {"CUDALongType", "torch.cuda.LongTensor"},
+          {"CUDAShortType", "torch.cuda.ShortTensor"},
+          {"CUDAHalfType", "torch.cuda.HalfTensor"},
+          {"CPUByteType", "torch.ByteTensor"},
+          {"CPUCharType", "torch.CharTensor"},
+          {"CPUDoubleType", "torch.DoubleTensor"},
+          {"CPUFloatType", "torch.FloatTensor"},
+          {"CPUIntType", "torch.IntTensor"},
+          {"CPULongType", "torch.LongTensor"},
+          {"CPUShortType", "torch.ShortTensor"},
+          {"CPUHalfType", "torch.HalfTensor"},
```

Comment:
a pattern-based find/replace could be even better -- this is `s/CUDA(Byte|Char|Double|Float|Int|Long|Short|Half)Type/torch.cuda.$1Tensor/` and such an approach wouldn't have to scan the string 7 additional times; similar comments apply for the other families of search strings.